src/drivers/intel/ssp/ssp.c

```diff
@@ -67,23 +67,44 @@ static void ssp_empty_tx_fifo(struct dai *dai)
 		ssp_write(dai, SSSR, sssr);
 }
 
+/*
+ * Empty SSP receive FIFO before DMA startup and
+ * ensure the DMA request is not already raised when
+ * DMA is started. If TX is not active, this is a no-op.
+ */
+static void ssp_empty_rx_fifo_for_start(struct dai *dai)
+{
+	uint32_t retry = SSP_RX_FLUSH_RETRY_MAX;
+	uint32_t entries;
+	uint32_t i;
+
+	while ((ssp_read(dai, SSSR) & SSSR_RNE) && retry--) {
+		entries = SSCR3_RFL_VAL(ssp_read(dai, SSCR3));
+
+		/* let DMA consume data or read RX FIFO directly */
+		for (i = 0; i < entries + 1; i++)
+			ssp_read(dai, SSDR);
+
+		entries = SSCR3_RFL_VAL(ssp_read(dai, SSCR3));
+	}
+
+	/* clear interrupt */
+	ssp_update_bits(dai, SSSR, SSSR_ROR, SSSR_ROR);
+
+	if (ssp_read(dai, SSSR) & SSSR_RFS)
+		dai_warn(dai, "ssp_empty_rx_fifo_for_startup(), RFS=1 after flush!");
+}
+
 /* empty SSP receive FIFO */
 static void ssp_empty_rx_fifo(struct dai *dai)
```

Comment:
if the other incarnation is `ssp_empty_rx_fifo_for_start()` then this is `ssp_empty_rx_fifo_for_stop()`?